workflow/controller/operator.go

```diff
@@ -754,6 +865,7 @@ func (woc *wfOperationCtx) markNodePhase(nodeName string, phase wfv1.NodePhase,
 	}
 	woc.wf.Status.Nodes[nodeID] = node
 	woc.updated = true
+	woc.log.Debugf("Marked node %s %s\n", nodeName, phase)
```

Comment:
Just heads up: in part of my dag branch, i've improved this method to only print status changes (as well as only mark update = true when something changes). This reduces log volume making debugging easier, and prevents unnecessary DB updates.